Texdiag/texdiag.cpp

```diff
@@ -1368,7 +1368,7 @@ namespace
     }
 
     //--------------------------------------------------------------------------------------
-#define SIGN_EXTEND(x,nb) ((((x)&(1<<((nb)-1)))?((~0)<<(nb)):0)|(x))
+#define SIGN_EXTEND(x,nb) ((((x)&(1<<((nb)-1)))?((~0)^((1<<(nb))-1)):0)|(x))
```

Comment:
This same macro appears in ``BC6HBC7.cpp``, but I don't see that file in the PR...

Reply:
Good catch. I think the reason it doesn't get flagged by clang-cl in `BC6HBC7.cpp` is because the additive-expression (RHS) is a variable `int(Prec.r))`, not a compile-time constant like it is here. PREFast still flags it though so I'll update it to match.